src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -364,11 +364,20 @@ public void EnsureCanonical(Block block)
             }
         }
 
-        private void RemoveBlockTx(Block block)
+        private void RemoveBlockTx(Block block, Block? exceptBlock = null)
         {
+            HashSet<Keccak> newTxs = null;
+            if (exceptBlock is not null)
+            {
+                newTxs = new HashSet<Keccak>(exceptBlock.Transactions.Select((tx) => tx.Hash));
+            }
+
```

Comment:
Just wondering if building HashSet is optimal here

Reply:
What do you have in mind?

Reply:
can we avoid allocation, what would have best performance

Reply:
Hmm.... Usually exceptBlock would be null, so I think its probably fine.